File: 973-k-closest-points-to-origin/973-k-closest-points-to-origin.js
/**
 * @param {number[][]} points
 * @param {number} k
 * @return {number[][]}
 */
var kClosest = function(points, k) {
    const sorted = points.sort((a,b)=>{
        return distance(a[0],a[1]) - distance(b[0],b[1]);
    });
    return sorted.slice(0,k);
};

function distance(x,y){
    return Math.sqrt(x*x+y*y);
}

File: 973-k-closest-points-to-origin/973-k-closest-points-to-origin.py
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        points_by_distance = dict()
        
        tuples = []
        for point in points:
            dist = math.sqrt( point[0]**2 + point[1]**2 )
        
            tuples.append((dist,point))
            
        heapq.heapify(tuples)

        res = []
        for i in range(k):
            res.append(heapq.heappop(tuples)[1])
        return res